desktop-environment.el

```diff
@@ -411,12 +421,28 @@ the screen."
 
 (declare-function exwm-input-set-key "ext:exwm-input")
 
-(defun desktop-environment-exwm-set-global-keybindings ()
-  "When using EXWM, add `desktop-environment-mode-map' to global keys."
+(defun desktop-environment-exwm-set-global-keybindings (enable)
+  "When using EXWM, add `desktop-environment-mode-map' to global keys if ENABLE
+  is non-nil.  When `desktop-environment-update-exwm-global-keys' is set to
+  :prefix, the bindings will be disabled when ENABLE is nil."
   (when (featurep 'exwm-input)
-    (map-keymap (lambda (event definition)
-                  (exwm-input-set-key (vector event) definition))
-                desktop-environment-mode-map)))
+    (cl-case desktop-environment-update-exwm-global-keys
+      (:global
+       (when enable
+         (map-keymap (lambda (event definition)
+                       (exwm-input-set-key (vector event) definition))
+                     desktop-environment-mode-map)))
+      (:prefix
+       (map-keymap (lambda (event definition)
+                     (ignore definition)
+                     (setq exwm-input-prefix-keys (if enable
+                                                      (cons event exwm-input-prefix-keys)
+                                                    (delq event exwm-input-prefix-keys))))
+                   desktop-environment-mode-map))
+      (nil nil)
```

Comment:
According to my tests, I think this should be `((nil) nil)` instead. The documentation says:

> Each clause looks like (KEYLIST BODY...).  EXPR is evaluated and compared against each key in each KEYLIST; the corresponding BODY is evaluated.  If no clause succeeds, cl-case returns nil.  A **single non-nil atom** may be used in place of a KEYLIST of one atom.  A KEYLIST of t or otherwise is allowed only in the final clause, and matches if no other keys match.  Key values are compared by eql.

Reply:
Of course. Will fix that.